gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/access/index/GfxdIndexManager.java

```diff
@@ -438,8 +416,15 @@ public void onEvent(final LocalRegion owner, final EntryEventImpl event,
               owner.getFullPath());
         }
 
+        boolean isRecovered = false;
+        if (!owner.isInitialized()) {
+          DiskRegion dr = owner.getDiskRegion();
+          if (dr != null) {
+            isRecovered = dr.testIsRecovered(entry, false);
+          }
+        }
         boolean throwEEE = false;
-        if (!posDup && owner.isInitialized()) {
+        if (!posDup && (owner.isInitialized() || (!owner.isInitialized() && !isRecovered))) {
```

Comment:
a) Other copies throws EEE, but in this VM the change gets applied and entry has different value. In future any update/select will cause issue.
b) ABRM#processSuspect adds it to suspect entries if if the entry is recovered. 

The issue was
1. GII put causes entry to be created and in put in ABRM
2. An invalid insert is coming at the same time on the uninitialized region
3. Suspect entries doesn't add that to suspect list but lets it proceed as per logic of "if the entry is already present in map". Either we should change that i.e. we should add it to suspect list so that when suspect list is processed it is ignored or throw EEE from onEvent.